apps/ui/modules/reservationUnit.ts

```diff
@@ -137,20 +141,15 @@ export function getReservationUnitName(
 
 export function getUnitName(
   unit?: Pick<UnitNode, "nameFi" | "nameSv" | "nameEn"> | null,
-  language: string = i18n?.language ?? "fi"
+  locale: string = i18n?.language ?? "fi"
 ): string | undefined {
   if (unit == null) {
     return undefined;
   }
-  const key = `name${capitalize(language)}`;
-  if (key in unit) {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any -- silly magic to avoid implicit any type
-    const val: unknown = (unit as any)[key];
-    if (typeof val === "string" && val.length > 0) {
-      return val;
-    }
-  }
-  return unit.nameFi ?? "-";
+  return (
+    getTranslationSafe(unit, "name", convertLanguageCode(locale ?? "")) ??
+    undefined
+  );
```

Comment:
Is this a part of the multi select refactor also, or "just" fixing stuff in general?